### PR TITLE
[wasm][debugger] Fixing assert while debugging.

### LINKED ISSUE #58021: [wasm][debugger] Assertion while debugging
image

* Assertion at /__w/1/s/src/mono/mono/metadata/class.c:5769, condition `<disabled>' not met

* Assertion at /__w/1/s/src/mono/mono/metadata/class.c:5769, condition `<disabled>' not met

* Assertion at /__w/1/s/src/mono/mono/metadata/class.c:5769, condition `<disabled>' not met

* Assertion at /__w/1/s/src/mono/mono/metadata/class.c:5769, condition `<disabled>' not met

### PR DESCRIPTION
**Pull request by thaystg**
When trying to evaluate DebuggerProxyAttribute of a generic type <T, K>, it was working only for one parameter <K>.
Fixes https://github.com/dotnet/runtime/issues/58021

### CODE REVIEW

**Review by lewing — APPROVED** [review]

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs [hunk 1]
```diff
@@ -2280,10 +2280,16 @@ public async Task<JArray> GetValuesFromDebuggerProxyAttribute(SessionId sessionI
...
+                            else
+                                typeToSearch += "]";
                         }
-                        typeToSearch += "]]";
+                        typeToSearch += "]";
                         typeToSearch +=  ", " + assemblyName;
                         var genericTypeId = await GetTypeByName(sessionId, typeToSearch, token);
+                        if (genericTypeId < 0)
+                            return null;
                         methodId = await GetMethodIdByName(sessionId, genericTypeId, ".ctor", token);
```
Comment (radical) [inline src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs:2293]: Won't this fail to find the correct .ctor, if there is more than one on the type?

Also, `DebuggerTypeProxy` accepts a string arg with the type name. That doesn't have to be in this PR though.

**Review by radical — APPROVED** [review]
My comments can be addressed in a separate PR. LGTM!

**Comment by lewing** [comment]
ci hit https://github.com/dotnet/runtime/issues/58062

**Pull request merged**